client/blocks/stripe-utils/normalize.js

```diff
@@ -0,0 +1,176 @@
+ *
+ * @return {StripePaymentItem[]} An array of PaymentItems
+ */
+const normalizeLineItems = ( cartTotalItems, pending = false ) => {
+	return cartTotalItems
+		.map( ( cartTotalItem ) => {
+			return cartTotalItem.value
+				? {
+						amount: cartTotalItem.value,
+						label: cartTotalItem.label,
+						pending,
+				  }
+				: false;
+		} )
+		.filter( Boolean );
+};
+
+/**
+ * Normalizes incoming cart shipping option items for use as shipping options
+ * with the Stripe api.
+ *
+ * @param {CartShippingOption[]}  shippingOptions An array of CartShippingOption items.
+ *
+ * @return {StripeShippingOption[]}  An array of Stripe shipping option items.
+ */
+const normalizeShippingOptions = ( shippingOptions ) => {
+	const rates = shippingOptions[ 0 ].shipping_rates;
+	return rates.map( ( rate ) => {
+		return {
+			id: rate.rate_id,
+			label: rate.name,
+			detail: rate.description,
+			amount: parseInt( rate.price, 10 ),
+		};
+	} );
+};
+
+/**
+ * Normalize shipping address information from stripe's address object to
+ * the cart shipping address object shape.
+ *
+ * @param {StripeShippingAddress} shippingAddress Stripe's shipping address item
+ *
+ * @return {CartShippingAddress} The shipping address in the shape expected by
+ * the cart.
+ */
+const normalizeShippingAddressForCheckout = ( shippingAddress ) => {
+	const address = {
+		first_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 0, 1 )
+			.join( ' ' ),
+		last_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 1 )
+			.join( ' ' ),
+		company: '',
+		address_1:
+			typeof shippingAddress.addressLine[ 0 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 0 ],
+		address_2:
+			typeof shippingAddress.addressLine[ 1 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 1 ],
+		city: shippingAddress.city,
+		state: shippingAddress.region,
+		country: shippingAddress.country,
+		postcode: shippingAddress.postalCode.replace( ' ', '' ),
+	};
+	return address;
+};
+
+/**
+ * Normalizes shipping option shape selection from Stripe's shipping option
+ * object to the expected shape for cart shipping option selections.
+ *
+ * @param {StripeShippingOption} shippingOption The customer's selected shipping
+ *                                              option.
+ *
+ * @return {string[]}  An array of ids (in this case will just be one)
+ */
+const normalizeShippingOptionSelectionsForCheckout = ( shippingOption ) => {
+	return shippingOption.id;
+};
+
+/**
+ * Returns the billing data extracted from the stripe payment response to the
+ * CartBillingData shape.
+ *
+ * @param {StripePaymentResponse} paymentResponse Stripe's payment response
+ *                                                object.
+ *
+ * @return {CartBillingAddress} The cart billing data
+ */
+const getBillingData = ( paymentResponse ) => {
+	const source = paymentResponse.source;
+	const name = source && source.owner.name;
+	const billing = source && source.owner.address;
```

Comment:
`source.owner` [may be `null` here](https://github.com/stripe/stripe-js/blob/db51d9576a66ff0de4326aeac19dc73934935bc8/types/api/Sources.d.ts#L87)

```suggestion
	const name = source?.owner?.name;
	const billing = source?.owner?.address;
```

Reply:
Fixed in d0afa3a79374e886e14b8946d147bfa509812051